### PR TITLE
Adding a custom form validation topic

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) provides a new topic, where it is described how to validate any custom form. 

## Affected DevDocs pages

N/A

## Links to Magento source code

-  [Initiating validation](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Backend/view/adminhtml/templates/admin/login.phtml#L12)
-  [Field validation using `data-validate` attribute](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Backend/view/adminhtml/templates/admin/login.phtml#L29)
- [Field validation using class names](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Backup/view/adminhtml/templates/backup/dialogs.phtml#L26-L28)

## Comments

Do you think that it makes sense to list all the available validation rules (or at least a part of them) here, based on these [rules.js](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js#L60)?
There are more than 80 of such rules, and the table may became a really big one. Maybe it makes sense to add just few of them?

whatsnew
Describing how to perform any kind of validation to a custom form.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+group: frontend-developer-guide
+title: Custom form validation
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+---
+
+This tutorial shows you how to validate your own form, using `mage/validation` library, before submitting it to the server.
+This gives the possibility to customer to fix all the invalid data straight away, which improves the good user experience.
```
Comment (rogyar) [inline src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md]: As a rule, we don't need to improve a good experience. Unless we are going to make it perfect 🙂

```suggestion
This gives the possibility to customer to fix all the invalid data straight away, which improves the user experience and user interface accessibility.
```

File: src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+### As `data-validate` attribute
+
+```html
+<input id="field-1" ... data-validate='{"required":true}'/>
+```
+
+### As attribute
+
+```html
+<input id="field-1" ... required="true"'/>
```
Comment (rogyar) [inline src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md]:
```suggestion
<input id="field-1" ... required="true"/>
```

File: src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+### As class name
+
+```html
+<input id="field-1" ... class="input-text required-entry"'/>
```
Comment (rogyar) [inline src/guides/v2.3/frontend-dev-guide/validations/custom-form-validation.md]:
```suggestion
<input id="field-1" ... class="input-text required-entry"/>
```

**Comment by rogyar** [comment]
As for your question about the existing rules 

> Do you think that it makes sense to list all the available validation rules (or at least a part of them) here, based on these [rules.js](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js#L60)?
> There are more than 80 of such rules, and the table may became a really big one. Maybe it makes sense to add just few of them?

On the one hand, describing them all might be a little bit overwhelming. On the other hand, many developers "Google: for the solutions. So having all validation rules indexed by a search engine may be beneficial. 

@dobooth would you, please, share your opinion on that? 

Thank you

**Comment by eduard13** [comment]
Hi @rogyar, @dobooth any updates on this one?
Thank you.

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
sorry for the delay!

**Comment by eduard13** [comment]
@dobooth but currently we are already linking it in `Form validation rules` section -> `All available Magento validation rules may be found in [validation/rules.js]`, or do you think we need to rewrite it?
Thank you.

**Comment by dobooth** [comment]
Ah right. Nevermind.

**Comment by dobooth** [comment]
running tests

**Comment by hguthrie** [comment]
@dobooth  Can you update the whatsnew with a complete sentence and a link to the new topic?

**Pull request merged**